### PR TITLE
Hook up missing link icons for linked and cloud smart objects

### PR DESCRIPTION
**Pull request by volfied**
Requires pgdv.679 or later.

layer.smartObject now contains `linkMissing` and `linkChanged` properties. We use `linkMissing` to change our own layer panel icons.

We currently do not get any notification from PS when this status changes, so it's up to us to later on hook up queries and model updates on our side.

Addresses #1990 and #2083

### CODE REVIEW

**Comment by designedbyscience** [comment]
@volfied I have some CSS and icon feedback...not sure if you want to deal with that or make some bugs for those changes instead. 

For the link changed, is it the case that while we will know on document open whether a link is out of date, we won't know if someone updates it?

**Comment by volfied** [comment]
@designedbyscience it would be great if you could grab the branch and make those changes yourself for the icons/style, as I just hooked up the already available icons. 

Yes, currently it only works on document open/update or layer reset to get the latest status. Due to way PS stores linked file data, there wasn't a simple way for us to get notifications on layers' linked info being updated.

**Comment by placegraphichere** [comment]
icon assets are in #2732

**Comment by iwehrman** [comment]
This is needed for UIF.

**Comment by designedbyscience** [comment]
Assigning this to @mcilroyc since he did the last round of Smart Object changes (thanks Cory!)

File: src/js/jsx/sections/layers/LayerFace.jsx [hunk 1]
```diff
@@ -315,7 +315,20 @@ define(function (require, exports, module) {
                         selected={!layer.visible}
                         onClick={this._handleVisibilityToggle}>
                     </ToggleButton>
-                );
+                ),
+                iconClassModifier;
+
+            if (layer.isSmartObject()) {
```
Comment (mcilroyc) [inline src/js/jsx/sections/layers/LayerFace.jsx]: idle comment: It bugs me that `isSmartObject` is a prototype function and not a cached property.  Compare to `superSelectable` in models/layer

**Comment by mcilroyc** [comment]
LGTM

**Pull request merged**